jobs/openshift/cluster/brew/Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+        options.clusterAdminUsername = userInput.clusterAdminUsername
+        options.clusterAdminPassword = userInput.clusterAdminPassword
+        options.installationGitUrl = userInput.installationGitUrl
+        options.installationGitBranch = userInput.installationGitBranch
+        verifyImageSyncOptions(options)
+    }
+}
+
+def setClusterAdminCredentials() {
+    def integreatlyCredentialsID = 'tower-openshift-cluster-credentials'
+    def clusterAdminCredentials = [:]
+    withCredentials([usernamePassword(credentialsId: integreatlyCredentialsID, usernameVariable: 'CLUSTER_ADMIN_USERNAME', passwordVariable: 'CLUSTER_ADMIN_PASSWORD')]) {
+        clusterAdminCredentials.clusterAdminUsername = "${CLUSTER_ADMIN_USERNAME}"
+        clusterAdminCredentials.clusterAdminPassword = "${CLUSTER_ADMIN_PASSWORD}"
+    }
+
+    return clusterAdminCredentials
+}
+
+String getHost(String url) {
+    URI uri = new URI(url);
+    String[] urlParts = uri.getHost().split(/\./).takeRight(3)
+    return urlParts.join('.')
+}
+
+stage('Image Sync Options') {
+    def clusterAdminCredentials = setClusterAdminCredentials()
+    imageSyncOptions.openshiftMasterUrl = params.openshiftMasterUrl
+    imageSyncOptions.clusterAdminUsername = params.clusterAdminUsername ?: clusterAdminCredentials.clusterAdminUsername // Defaults to the username stored in tower-openshift-cluster-credentials
+    imageSyncOptions.clusterAdminPassword = params.clusterAdminPassword ?: clusterAdminCredentials.clusterAdminPassword // Defaults to the password stored in tower-openshift-cluster-credentials
+    imageSyncOptions.installationGitUrl = params.installationGitUrl
+    imageSyncOptions.installationGitBranch = params.installationGitBranch
+    verifyImageSyncOptions(imageSyncOptions)
+    
+    println "Image Sync Options: ${imageSyncOptions}"
+    currentBuild.displayName = "${currentBuild.displayName} ${imageSyncOptions.installationGitBranch}"
+    currentBuild.description = imageSyncOptions.openshiftMasterUrl
+}
+
+def githubSSHCredentialsID = 'jenkinsgithub'
+def imageSyncConfigFile = './integreatly.yml'
+String brewContainerRegistry
+def projectsToSync
+def syncImages = false
+
+node('cirhos_rhel7') {
+    cleanWs()
+    stage('Ensure Sync File Exists') {
+        dir('installation') {
+            gitCheckoutRepo(imageSyncOptions.installationGitUrl, imageSyncOptions.installationGitBranch, githubSSHCredentialsID, '.')
+            syncImages = fileExists imageSyncConfigFile
+
+            if (!syncImages) {
+                println "Image sync file '${imageSyncConfigFile}' is not available. No images to sync, skipping!"
+            }
+        }
+    }
+
+    stage('Get Images To Sync') {
+        when(syncImages) {
+            dir('installation') {
+                def imageSyncConfigContent = readYaml file: imageSyncConfigFile
+                projectsToSync = imageSyncConfigContent.imageSync
+                brewContainerRegistry = imageSyncConfigContent.sourceRepo
```

Comment:
This should allow a sourceRepo to be set at the higher level right?

```

imageSync:
  sourceRepo: brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888
  fuse7:
    images:
    - fuse-ignite-meta:1.3-16
```

Reply:
I was thinking it would be on the same level as `imageSync`. That would make it easier to iterate through the list of projects without having to ignore `sourceRepo`. But if there would be other stuff in this file apart form imageSync, then it makes sense to have that in there.